clients/grpc-client/src/main/java/org/projectnessie/client/grpc/GrpcExceptionMapper.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.grpc;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.google.common.base.Throwables;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import java.security.AccessControlException;
+import java.util.concurrent.Callable;
+import javax.ws.rs.WebApplicationException;
+import org.jboss.resteasy.api.validation.ResteasyViolationException;
+import org.projectnessie.client.rest.NessieBadRequestException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieError;
+import org.projectnessie.error.NessieNotFoundException;
+
+/** Maps gRPC exceptions to Nessie-specific exceptions and the other way around. */
+public class GrpcExceptionMapper {
+
+  /**
+   * Takes the given Exception and converts it to a {@link StatusRuntimeException} with a specific
+   * gRPC Status code.
+   *
+   * @param ex The exception to convert to a {@link StatusRuntimeException} with a specific gRPC
+   *     Status code.
+   * @return A new {@link StatusRuntimeException} with a specific gRPC status code.
+   */
+  public static StatusRuntimeException toProto(Exception ex) {
+    if (ex instanceof NessieNotFoundException) {
+      return Status.NOT_FOUND.withDescription(ex.getMessage()).withCause(ex).asRuntimeException();
+    }
+
+    if (ex instanceof NessieConflictException) {
+      return Status.ALREADY_EXISTS
+          .withDescription(ex.getMessage())
+          .withCause(ex)
+          .asRuntimeException();
+    }
+
+    if (ex instanceof JsonParseException
+        || ex instanceof JsonMappingException
+        || ex instanceof IllegalArgumentException
+        || ex instanceof WebApplicationException) {
+      return Status.INVALID_ARGUMENT
+          .withDescription(ex.getMessage())
+          .withCause(ex)
+          .asRuntimeException();
+    }
+    if (ex instanceof ResteasyViolationException) {
```

Comment:
nit: style: we have new lines before `if` is some cases here, and no new line in others

Reply:
fixed